### PR TITLE
feat: Replay scrub preview

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

In a big recording it's often nicer to be able to skip through to see a preview of what is coming up. We already have a timestamp preview but we could also do a real preview.

## Changes

* Refactors the playerlogics so that we can have multiple running that don't interfere with each other
* Adds a preview component of another PlayerFrame
* NOTE: This doesn't solve the long-buffer problem and it is not super performant

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/lib/utils.tsx [hunk 1]
```diff
@@ -1677,3 +1677,14 @@ export function shouldCancelQuery(error: any): boolean {
     // the query will continue running in ClickHouse
     return error.name === 'AbortError' || error.message?.name === 'AbortError' || error.status === 504
 }
+
+export function debounce<F extends (...args: Parameters<F>) => ReturnType<F>>(
```
Comment (daibhin) [inline frontend/src/lib/utils.tsx:1681]: Would the [`use-debounce` package](https://github.com/xnimorz/use-debounce#debounced-callbacks) we already have installed handle this?

**Review by pauldambra — APPROVED** [review]
I think the debounce might be too aggressive while I'm moving the mouse I'd still expect the view to change



But this is behind a flag and does what it says on the tin!

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts [hunk 2]
```diff
@@ -195,43 +192,45 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
                 loadRecordingMetaFailure: () => true,
             },
         ],
+        snapshotsLoaded: [
+            false as boolean,
+            {
+                loadRecordingSnapshotsSuccess: () => true,
+                loadRecordingSnapshotsFailure: () => true,
```
Comment (daibhin) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:199]: Should this be `false`?

File: frontend/src/lib/utils.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/lib/utils.tsx:1681]: Weird. I searched and didn't find it. Looks like exactly what I need though 👍

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts (hunk 2 above)
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:199]: `loadSnapshotsTried` `loadSnapshotsStarted` ??

**Pull request merged**